lib/scope.js

```diff
@@ -103,11 +103,15 @@ AST_Toplevel.DEFMETHOD("figure_out_scope", function(options){
     var nesting = 0;
     var in_destructuring = null;
     var tw = new TreeWalker(function(node, descend){
-        if (options.screw_ie8 && node instanceof AST_Catch) {
+        var create_a_block_scope =
```

Comment:
Would it be useful to create a method on `AST_Block` instead of handling all the cases here?